templates/index.html

```diff
@@ -2,17 +2,96 @@
 {% block title %}To-Do App{% endblock %}
 
 {% block content %}
-  <div class="jumbotron">
-    <h1 class="display-4">To-Do App</h1>
-    <p class="lead">Just another to-do app.</p>
-  </div>
+<div class="jumbotron">
+  <h1 class="display-4">To-Do App</h1>
+  <p class="lead">Just another to-do app.</p>
+</div>
 
-  <div class="row">
-    <div class="col-md-12">
-      <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
-    </div>
+<form class="form-inline" action="/create" method="POST">
+  <div class="form-group mx-sm-3 mb-2">
+    <input type="text" class="form-control" name="newItemTitle" id="newItemTitle" placeholder="New Item Title...">
+  </div>
+  <div class="form-group mb-2">
+    <input type="submit" value="Create" class="btn btn-primary">
   </div>
+</form>
+
+<div class="row">
+  <div class="col-md-12">
+    <h2 align=center>Items</h2>
+    <table class="table table-hover">
+      <thead>
+        <tr>
+        <th class="th-sm">
+          <a href="{{ url_for('sort_items', sortType='title') }}">
+            Title
+          </a>
+        </th>
+        <th scope="col">
+          <a href="{{ url_for('sort_items', sortType='status') }}">
+            Status
+          </a>
+        </th>
+        <th scope="col" />
+        </tr>
+      </thead>
+      <tbody>
+        {% for item in items %}
+        <tr>
+          <td>{{ item.title }}</td>
+          <td>{{ item.status }}</td>
+          <td>
+            <button type="button" class="btn btn-primary" data-toggle="modal" data-target='#itemModal-{{ item.id }}'>
+              Edit
+            </button>
+          </td>
+        </tr>
+
+        <div class="modal" tabindex="-1" role="dialog" id='itemModal-{{ item.id }}'>
```

Comment:
For a javascriptless approach I'd move the modal to a separate page and keep the for loop short and simple (but this is a very minor point)